addons/point_of_sale/static/src/app/models/pos_order.js

```diff
@@ -1060,6 +1062,43 @@ export class PosOrder extends Base {
     getFloatingOrderName() {
         return this.note || this.tracking_number;
     }
+
+    sortBySequenceAndCategory(a, b) {
+        const seqA = a.product_id?.pos_categ_ids[0]?.sequence ?? 0;
+        const seqB = b.product_id?.pos_categ_ids[0]?.sequence ?? 0;
+        const pos_categ_id_A = a.product_id?.pos_categ_ids[0]?.id ?? 0;
+        const pos_categ_id_B = b.product_id?.pos_categ_ids[0]?.id ?? 0;
+
+        if (seqA !== seqB) {
+            return seqA - seqB;
+        }
+        return pos_categ_id_A - pos_categ_id_B;
+    }
+
+    // orderlines will be sorted on the basis of pos product category and sequence for group the orderlines in order cart
+    getSortedOrderlines() {
+        if (this.config.orderlines_sequence_in_cart_by_category && this.lines.length) {
+            const linesToSort = [...this.lines];
+            linesToSort.sort(this.sortBySequenceAndCategory);
+            const resultLines = [];
+            linesToSort.forEach((line) => {
+                if (line.combo_line_ids?.length > 0) {
+                    resultLines.push(line);
+                    const childLines = linesToSort.filter(
+                        (childLine) => childLine.combo_parent_id?.id === line.id
+                    );
+                    childLines
+                        .sort(this.sortBySequenceAndCategory)
+                        .forEach((childLine) => resultLines.push(childLine));
```

Comment:
```suggestion
                    const childLines = linesToSort.filter(
                        (childLine) => childLine.combo_parent_id?.id === line.id
                    );
                    resultLines.push(...childLines);
```

I don't think you need to sort the childLines again since they are already sorted (derived from `linesToSort`).